.github/workflows/python-so-copying.yml

```diff
@@ -22,18 +22,19 @@ defaults:
 
 jobs:
   build:
-    runs-on: ubuntu-latest
     name: "TILEDB_EXISTS: ${{ matrix.TILEDB_EXISTS }} TILEDBSOMA_EXISTS: ${{ matrix.TILEDBSOMA_EXISTS }}"
     strategy:
       fail-fast: false
       matrix:
+        os: [ubuntu-22.04, macos-12]
         TILEDB_EXISTS: ["no", "yes"]
         TILEDBSOMA_EXISTS: ["no", "yes"]
         exclude:
           - TILEDB_EXISTS: "no"
             TILEDBSOMA_EXISTS: "yes"
+    runs-on: ${{ matrix.os }}
     container:
-      image: ubuntu:22.04
+      image: ${{ matrix.os }}
```

Comment:
This is the tricky part. This needs to be a Docker image, not the name of the GitHub Actions runner.

The linux job runs inside of a Docker container (this made it easier for me to test locally). Thus it's not as easy as adding `macos-12` to the build matrix. We need to either 1) make the container directive only apply to the  ubuntu job (probably possible, but not immediately clear to me how), or 2) create a second job that runs similar code on macOS (which might be worth it given all the changes to accommodate running the same steps on linux and macOS)

Reply:
@jdblischak I'm totally happy with well-documented code-duplication here.

Reply:
I did find some [workarounds](https://stackoverflow.com/questions/76919426/is-it-possible-to-use-docker-for-linux-in-a-matrix-of-operating-systems) to create a build matrix that builds natively on macOS but in Docker on Ubuntu. There is no `if`, so the key is to pass `null` to `container` for the macOS job. Overall the readability is not great though.

Also, I realized that we'll also need to download a different pre-built libtiledb binary for the macOS build, which makes the option to have a completely separate macOS build all the more appealing.